patches/chromium/fix_patch_out_profile_refs_in_accessibility_ui.patch

```diff
@@ -0,0 +1,139 @@
+   // command line switch has been used.
+-  bool are_accessibility_image_labels_enabled =
+-      is_web_enabled &&
+-      pref->GetBoolean(prefs::kAccessibilityImageLabelsEnabled);
++  bool are_accessibility_image_labels_enabled = is_web_enabled;
+   bool label_images = mode.has_mode(ui::AXMode::kLabelImages);
+   data.SetString(kLabelImages, are_accessibility_image_labels_enabled
+                                    ? (label_images ? kOn : kOff)
+@@ -209,7 +205,7 @@ void HandleAccessibilityRequestCallback(
+   // The "pdf" flag is independent of the others.
+   data.SetString(kPDF, pdf ? kOn : kOff);
+ 
+-  bool show_internal = pref->GetBoolean(prefs::kShowInternalAccessibilityTree);
++  bool show_internal = true;
+   data.SetString(kInternal, show_internal ? kOn : kOff);
+ 
+   std::unique_ptr<base::ListValue> rvh_list(new base::ListValue());
+@@ -244,15 +240,13 @@ void HandleAccessibilityRequestCallback(
+     rvh_list->Append(std::move(descriptor));
+   }
+   data.Set(kPagesField, std::move(rvh_list));
+-
+   std::unique_ptr<base::ListValue> browser_list(new base::ListValue());
+-#if !defined(OS_ANDROID)
++#if 0
+   for (Browser* browser : *BrowserList::GetInstance()) {
+     browser_list->Append(BuildTargetDescriptor(browser));
+   }
+-#endif  // !defined(OS_ANDROID)
++#endif // !defined(OS_ANDROID)
+   data.Set(kBrowsersField, std::move(browser_list));
+-
+   std::string json_string;
+   base::JSONWriter::Write(data, &json_string);
+ 
+@@ -477,9 +471,11 @@ void AccessibilityUIMessageHandler::SetGlobalFlag(const base::ListValue* args) {
+ 
+   AllowJavascript();
+   if (flag_name_str == kInternal) {
++#if 0
+     PrefService* pref = Profile::FromWebUI(web_ui())->GetPrefs();
+     pref->SetBoolean(prefs::kShowInternalAccessibilityTree, enabled);
+     return;
++#endif
+   }
+ 
+   ui::AXMode new_mode;
+@@ -581,13 +577,14 @@ void AccessibilityUIMessageHandler::RequestWebContentsTree(
+       content::AccessibilityTreeFormatter::PropertyFilter::ALLOW_EMPTY);
+   AddPropertyFilters(property_filters, deny,
+                      content::AccessibilityTreeFormatter::PropertyFilter::DENY);
+-
++#if 0
+   PrefService* pref = Profile::FromWebUI(web_ui())->GetPrefs();
+   bool internal = pref->GetBoolean(prefs::kShowInternalAccessibilityTree);
+   base::string16 accessibility_contents_utf16 =
+       web_contents->DumpAccessibilityTree(internal, property_filters);
+   result->SetString(kTreeField,
+                     base::UTF16ToUTF8(accessibility_contents_utf16));
++#endif
+   CallJavascriptFunction(request_type, *(result.get()));
+ }
+ 
+@@ -626,7 +623,8 @@ void AccessibilityUIMessageHandler::RequestNativeUITree(
+       content::AccessibilityTreeFormatter::PropertyFilter::ALLOW_EMPTY);
+   AddPropertyFilters(property_filters, deny,
+                      content::AccessibilityTreeFormatter::PropertyFilter::DENY);
+-
++#endif
++#if 0
+   for (Browser* browser : *BrowserList::GetInstance()) {
```

Comment:
Electron's `WindowList` is synonym to `BrowserList`, can you swap it in so that `kTreeField` can be populated.

Reply:
@deepak1556 i considered this, but would that not also mean we would need to add another implementation of `BuildTargetDescriptor`?

Reply:
well its only a couple of lines, I don't think there is a drawback to it. On the other hand, this will allow to print the native ui tree.

```
std::unique_ptr<base::DictionaryValue> BuildTargetDescriptor(electron::NativeWIndow* window) {
  std::unique_ptr<base::DictionaryValue> target_data(
      new base::DictionaryValue());
  target_data->SetInteger(kIdField, window->GetId());
  target_data->SetString(kNameField, window->GetTitle());
  target_data->SetString(kTypeField, kElectronWindow);
  return target_data;
}
```